core/src/main/java/io/questdb/cairo/vm/MemoryPARWImpl.java

```diff
@@ -465,193 +235,512 @@ public final void putLong128(long l1, long l2) {
     }
 
     @Override
-    public void putLong256(long offset, Long256 value) {
-        putLong256(
-                offset,
-                value.getLong0(),
-                value.getLong1(),
-                value.getLong2(),
-                value.getLong3()
-        );
-    }
+    public final void putLong256(long l0, long l1, long l2, long l3) {
+        if (pageHi - appendPointer > Long256.BYTES - 1) {
+            Unsafe.getUnsafe().putLong(appendPointer, l0);
+            Unsafe.getUnsafe().putLong(appendPointer + Long.BYTES, l1);
+            Unsafe.getUnsafe().putLong(appendPointer + Long.BYTES * 2, l2);
+            Unsafe.getUnsafe().putLong(appendPointer + Long.BYTES * 3, l3);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).